academy/exchange/cloud/app.py

```diff
@@ -108,6 +108,7 @@ async def _share_mailbox_route(request: Request) -> Response:
         group_id = data['group_id']
 
     except (KeyError, ValidationError):
+        logger.warning('Missing or invalid mailbox id in request.')
```

Comment:
make this a `logger.exception` or add `exc_info=True` -- those exception objects might contain useful information. likewise for other warnings in exception contexts in this PR